integration_tests/auction_test.go

```diff
@@ -0,0 +1,239 @@
+		auctionQuery := types.QueryActiveAuctionRequest{
+			AuctionId: uint32(1),
+		}
+		auctionResponse, err := auctionQueryClient.QueryActiveAuction(context.Background(), &auctionQuery)
+		s.Require().NoError(err)
+		s.Require().Equal(auctionResponse.Auction.Id, uint32(1))
+		s.T().Log(auctionResponse.Auction)
+
+		bankQueryClient := banktypes.NewQueryClient(val0ClientCtx)
+		balanceRes, err := bankQueryClient.AllBalances(context.Background(), &banktypes.QueryAllBalancesRequest{Address: authtypes.NewModuleAddress(types.ModuleName).String()})
+		s.Require().NoError(err)
+		s.T().Log(balanceRes)
+
+		balanceRes, err = bankQueryClient.AllBalances(context.Background(), &banktypes.QueryAllBalancesRequest{Address: s.chain.orchestrators[0].keyInfo.GetAddress().String()})
+		s.Require().NoError(err)
+		s.T().Log(balanceRes)
+
+		bidRequest1 := types.MsgSubmitBidRequest{
+			AuctionId:              uint32(1),
+			Bidder:                 s.chain.orchestrators[0].keyInfo.GetAddress().String(),
+			MaxBidInUsomm:          sdk.NewCoin("usomm", sdk.NewIntFromUint64(5000)),
+			SaleTokenMinimumAmount: sdk.NewCoin("gravity0x3506424f91fd33084466f402d5d97f05f8e3b4af", sdk.NewIntFromUint64(1)),
+			Signer:                 s.chain.orchestrators[0].keyInfo.GetAddress().String(),
+		}
+
+		submitBid, err := s.chain.sendMsgs(*orchClientCtx, &bidRequest1)
+		s.Require().NoError(err)
+		s.T().Log(submitBid)
+
+		auctionResponse, err = auctionQueryClient.QueryActiveAuction(context.Background(), &auctionQuery)
+		s.Require().NoError(err)
+		s.T().Log(auctionResponse.Auction)
+
+		// Verify auction updated as expected
+		expectedAuction := types.Auction{
+			Id:                         uint32(1),
+			StartingTokensForSale:      sdk.NewCoin("gravity0x3506424f91fd33084466f402d5d97f05f8e3b4af", sdk.NewInt(5000)),
+			StartBlock:                 uint64(0),
+			EndBlock:                   uint64(0),
+			InitialPriceDecreaseRate:   sdk.MustNewDecFromStr("0.05"),
+			CurrentPriceDecreaseRate:   sdk.MustNewDecFromStr("0.05"),
+			PriceDecreaseBlockInterval: uint64(1000),
+			InitialUnitPriceInUsomm:    sdk.MustNewDecFromStr("2"),
+			CurrentUnitPriceInUsomm:    sdk.MustNewDecFromStr("2"),
+			RemainingTokensForSale:     sdk.NewCoin("gravity0x3506424f91fd33084466f402d5d97f05f8e3b4af", sdk.NewInt(2500)),
+			FundingModuleAccount:       cellarfees.ModuleName,
+			ProceedsModuleAccount:      cellarfees.ModuleName,
+		}
+		s.Require().Equal(expectedAuction, *auctionResponse.Auction)
+
+		// Verify user has funds debited and purchase credited
+		balanceRes, err = bankQueryClient.AllBalances(context.Background(), &banktypes.QueryAllBalancesRequest{Address: s.chain.orchestrators[0].keyInfo.GetAddress().String()})
+		s.Require().NoError(err)
+		s.T().Log(balanceRes)
+
+		s.Require().Equal(sdk.NewCoin("gravity0x3506424f91fd33084466f402d5d97f05f8e3b4af", sdk.NewInt(2500)), balanceRes.Balances[0])
+		s.Require().Equal(sdk.NewCoin("usomm", sdk.NewInt(99999995000)), balanceRes.Balances[2])
```

Comment:
Could you instead query the starting balance before and then require it to be `startingBalance.Amount.Sub(sdk.NewInt(5000))`? Other tests may (and mine do) add balances in genesis that break this

Reply:
Yup, will update